XenModel/Actions/Network/ChangeNetworkingAction.cs

```diff
@@ -195,6 +201,13 @@ private void ReconfigureManagement(bool this_host, int hi)
                                                           clearDownManagementIP);
         }
 
+        private void PoolReconfigureManagement(int hi)
+        {
+            System.Diagnostics.Trace.Assert(newManagement != null);

```

Comment:
Thanks for the suggestion @MihaelaStoica , I will incorporate the same. Your suggesting having an assertion inside an if statement for newManagement like if (newManagement == null) then assert ?